File: challenges/83/solution.ts
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */

// empty list
// can be negative, between -100 and 100
// ascending

/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function (head) {
  if (!head) return head;

  let currentNode = head;

  while (!!currentNode?.next) {
    while (currentNode.val === currentNode.next?.val || null) {
      currentNode.next = currentNode.next.next;
    }
    currentNode = currentNode.next;
  }
  return head;
};

File: challenges/83/solution.py
from typing import Optional

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:

    # edge case: []
    if head is None:
        return head

    node = head
    while(node is not None):

        # if the current node is the last one, finish
        if node.next is None:
            break

        # if the next node has the same value as the current one
        # exclude the next node from the list
        if node.val == node.next.val:
            node.next = node.next.next
        
        # otherwise, move to the next node
        else:
            node = node.next

    return head

    


    